internal/blockchain/command/mine_block.go

```diff
@@ -0,0 +1,93 @@
+package command
+
+import (
+	"github.com/patrykferenc/eecoin/internal/blockchain/domain/blockchain"
+	ev "github.com/patrykferenc/eecoin/internal/common/event"
+	"github.com/patrykferenc/eecoin/internal/transaction/domain/transaction"
+	"log/slog"
+	"time"
+)
+
+type MineBlock struct {
+	InterruptChannel chan bool
+}
+
+type MineBlockHandler interface {
+	Handle(cmd MineBlock)
+}
+
+type mineBlockHandler struct {
+	selfAddr       string
+	repository     BlockChainRepository
+	publisher      ev.Publisher
+	poolRepository transaction.PoolRepository
+}
+
+func NewMineBlockHandler(selfAddress string, repository BlockChainRepository, publisher ev.Publisher, poolRepository transaction.PoolRepository) MineBlockHandler {
+	return &mineBlockHandler{
+		repository:     repository,
+		publisher:      publisher,
+		poolRepository: poolRepository,
+		selfAddr:       selfAddress,
+	}
+}
+
+func (h *mineBlockHandler) Handle(cmd MineBlock) {
+	var chain = h.repository.GetChain()
+	var previousBlock = chain.GetLast()
+	var c, err = blockchain.NewChallenge(previousBlock.Challenge.Difficulty, previousBlock.Challenge.TimeCapMillis)
+
+	if err != nil {
+		slog.Error("Error creating challenge", "error", err)
+	}
+	for {
+		currentTime := time.Now().UnixMilli()
+		var transactions = h.poolRepository.GetAll()
+		err := c.RollNonce(previousBlock, transactions, currentTime)
+		if err != nil {
+			slog.Error("Error rolling nonce", "error", err)
+			continue
+		}
+		if c.MatchesDifficulty() {
+			tx, err := transaction.NewCoinbase(h.selfAddr, 10)
```

Comment:
I think we should roll also using coinbase

Reply:
![Uploading image.png…]()